src/libraries/Microsoft.Extensions.Configuration.Json/src/JsonConfigurationFileParser.cs

```diff
@@ -15,6 +15,7 @@ internal sealed class JsonConfigurationFileParser
 
         private readonly Dictionary<string, string?> _data = new Dictionary<string, string?>(StringComparer.OrdinalIgnoreCase);
         private readonly Stack<string> _paths = new Stack<string>();
+        private string _keyDelimiter = "`";// ConfigurationPath.KeyDelimiter;
```

Comment:
Of course, by changing `:` to `` ` ``, means we can't have `` ` `` in keys.
And obviously, this will need to be configurable, with a default defaulting to `:`.

Reply:
My other idea was for each `IConfigurationProvider` to specify its delimiter, but that would be a breaking change for older runtimes that don't support default interface implementation:
```csharp
    public interface IConfigurationProvider
    {
        public string GetDelimiter() => ":";
        ...
```